url901cc5721b_shidasan_KonohaCoffee_tgz-pJ8-org_KonohaScript_KonohaJ8/src/org/KonohaScript/KMethod.java

```diff
@@ -0,0 +1,106 @@
+ * Copyright (c) 2013, the Konoha project authors. All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***************************************************************************/
+
+package org.KonohaScript;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public class KMethod implements KonohaParserConst {
+	
+	int MethodFlag;
+	KClass ClassInfo;
+	String MethodName;
+//	int MethodSymbol;
+//	int CanonicalSymbol;
+	KParam Param;
+	Method MethodRef;
+
+//	String Source;
+//	long uline;
+//	KNameSpace LazyCompileNameSpace;
+	
+	public KMethod(int MethodFlag, KClass ClassInfo, String MethodName, KParam Param, Method MethodRef) {
+		this.MethodFlag = MethodFlag;
+		this.ClassInfo = ClassInfo;
+		this.MethodName = MethodName;
+		this.Param = Param;
+		this.MethodRef = MethodRef;	
+	}
+	
+	public boolean Match(String MethodName, int ParamSize /*, int DataSize, KClass[] ParamData */) {
+		if(MethodName.equals(this.MethodName)) {
+			if(Param.GetParamSize() == ParamSize) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	boolean IsStaticInvocation() {
+		return Modifier.isStatic(MethodRef.getModifiers());
+	}
+
+//	static {
+//		KClass IntClass.AddMethod(flag, "+", Param);
+//		KParam Param = KParam.ParseOf(ns, "int int x, int y");
+//		IntClass.Add(new KMethod(flag, IntClass, "+", Param, KFunc.LookupMethod(this, MethodName));
+//	}
+
+	public Object Eval(Object[] ParamData) {
+		int ParamSize = Param.GetParamSize();
+		try {
+			if(IsStaticInvocation()) {
+				switch(ParamSize) {
+				case 0: return MethodRef.invoke(null, ParamData[0]);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)